src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import {
+  Popover as CorePopover,
+  PopoverProps as CorePopoverProps,
+} from 'wix-ui-core/popover';
+
+import { TPAComponentProps } from '../../types';
+import { generateKey } from '../../common/random';
+import { KEYS } from '../../common/keyCodes';
+
+import { TriggerAction } from './constants';
+import { st, classes } from './Popover.st.css';
+
+export interface PopoverProps
+  extends TPAComponentProps,
+    Omit<CorePopoverProps, 'shown'> {
+  triggerAction?: TriggerAction;
+  wiredToSiteColors?: boolean;
+}
+
+export const Popover: React.FC<PopoverProps> & {
+  Element?: React.FC;
+  Content?: React.FC;
+} = props => {
+  const {
+    triggerAction,
+    wiredToSiteColors,
+    children,
+    className,
+    ...rest
+  } = props;
+  const [isShown, setIsShown] = React.useState<boolean>(false);
+  const [id] = React.useState(generateKey('popover'));
+
+  const [Element, Content] = React.Children.toArray(
+    children,
+  ) as React.ReactElement[];
+
+  return (
+    <CorePopover
+      id={id}
+      showArrow
+      shown={isShown}
+      className={st(classes.root, { wired: wiredToSiteColors }, className)}
+      {...getCoreProps()}
+      {...rest}
+    >
+      {React.cloneElement(
+        Element,
+        {},
+        React.cloneElement(Element.props.children, getTriggerProps()),
+      )}
+      {Content}
+    </CorePopover>
+  );
+
+  function getTriggerProps() {
```

Comment:
We prefer to use the `_` prefix for our internal methods:
```suggestion
  function _getTriggerProps() {
```
Please adjust the rest as well 🙏